microsoft_sentinel: Add agentless deployment

kcreddy opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
-->

## Proposed commit message

<!-- Mandatory
Explain here the changes you made on the PR.

Please explain:

- WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
- WHY:  the rationale/motivation for the changes

This text will be pasted into the squash dialog when the change is committed and will be
a long term historical record of the change to help future contributors understand the
change, please help them by making it clear and comprehensive, they may be you.

If the commit title is adequate to describe both of these things, The text here may be omitted
or replaced with "See title". The title of the PR will be used as the commit message title when
the merge is made and the "See title" marker will be removed if present.

The text here and the PR title will be subject to the PR review process.
-->

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

<!-- Recommended
Add a checklist of things that are required to be reviewed in order to have the PR approved
-->
- [ ] 

## How to test this PR locally

<!-- Recommended
Explain here how this PR will be tested by the reviewer: commands, dependencies, steps, etc.
-->

## Related issues

<!-- Recommended
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.

- Closes #123
- Relates #123
- Requires #123
- Supersedes #123
-->
- 

## Screenshots

<!-- Optional
Add here screenshots presenting:
- Kibana UI forms presenting configuration options exposed by the integration
- dashboards with collected metrics or logs
-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/21543)
* Commit: 33a051c532f931067332dc21173dd53412f49c06

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/21543#0194cf68-c9d7-48a3-94e9-b35f97e07bcd)

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21543","number":21543,"commit":"33a051c532f931067332dc21173dd53412f49c06"}],"number":21543}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/21568)
* Commit: 26e5fdeaa77f369e7cc9b67cce0a89363f267aa2

### History
* :broken_heart: [Build #21543](https://buildkite.com/elastic/integrations/builds/21543) failed 33a051c532f931067332dc21173dd53412f49c06

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21568","number":21568,"commit":"26e5fdeaa77f369e7cc9b67cce0a89363f267aa2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21543","number":21543,"commit":"33a051c532f931067332dc21173dd53412f49c06"}],"number":21568}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/21746)
* Commit: 384ba0aa4385e56775d1d0e167c96b046c3acd7d

### History
* :green_heart: [Build #21568](https://buildkite.com/elastic/integrations/builds/21568) succeeded 26e5fdeaa77f369e7cc9b67cce0a89363f267aa2
* :broken_heart: [Build #21543](https://buildkite.com/elastic/integrations/builds/21543) failed 33a051c532f931067332dc21173dd53412f49c06

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21746","number":21746,"commit":"384ba0aa4385e56775d1d0e167c96b046c3acd7d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21568","number":21568,"commit":"26e5fdeaa77f369e7cc9b67cce0a89363f267aa2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21543","number":21543,"commit":"33a051c532f931067332dc21173dd53412f49c06"}],"number":21746}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/security-service-integrations (Team:Security-Service Integrations)

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/21762)
* Commit: f059f98581936ff4e75b49e6cfdbb4f60fd5ef01

### History
* :green_heart: [Build #21746](https://buildkite.com/elastic/integrations/builds/21746) succeeded 384ba0aa4385e56775d1d0e167c96b046c3acd7d
* :green_heart: [Build #21568](https://buildkite.com/elastic/integrations/builds/21568) succeeded 26e5fdeaa77f369e7cc9b67cce0a89363f267aa2
* :broken_heart: [Build #21543](https://buildkite.com/elastic/integrations/builds/21543) failed 33a051c532f931067332dc21173dd53412f49c06

cc @kcreddy

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21762","number":21762,"commit":"f059f98581936ff4e75b49e6cfdbb4f60fd5ef01"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21746","number":21746,"commit":"384ba0aa4385e56775d1d0e167c96b046c3acd7d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21568","number":21568,"commit":"26e5fdeaa77f369e7cc9b67cce0a89363f267aa2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21543","number":21543,"commit":"33a051c532f931067332dc21173dd53412f49c06"}],"number":21762}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/21900)
* Commit: 393c46a1a7cab5a896a50c6ec295ffdbd726e3cf

### History
* :green_heart: [Build #21762](https://buildkite.com/elastic/integrations/builds/21762) succeeded f059f98581936ff4e75b49e6cfdbb4f60fd5ef01
* :green_heart: [Build #21746](https://buildkite.com/elastic/integrations/builds/21746) succeeded 384ba0aa4385e56775d1d0e167c96b046c3acd7d
* :green_heart: [Build #21568](https://buildkite.com/elastic/integrations/builds/21568) succeeded 26e5fdeaa77f369e7cc9b67cce0a89363f267aa2
* :broken_heart: [Build #21543](https://buildkite.com/elastic/integrations/builds/21543) failed 33a051c532f931067332dc21173dd53412f49c06

cc @kcreddy

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21900","number":21900,"commit":"393c46a1a7cab5a896a50c6ec295ffdbd726e3cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21762","number":21762,"commit":"f059f98581936ff4e75b49e6cfdbb4f60fd5ef01"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21746","number":21746,"commit":"384ba0aa4385e56775d1d0e167c96b046c3acd7d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21568","number":21568,"commit":"26e5fdeaa77f369e7cc9b67cce0a89363f267aa2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21543","number":21543,"commit":"33a051c532f931067332dc21173dd53412f49c06"}],"number":21900}
buildkite-pr-comment-->

kcreddy commented:
Thanks for reviewing @chemamartinez 😄 
We would like to run some smoke-tests with valid Microsoft Sentinel credentials. So just waiting for them and other reviews before merging.

qcorporation commented:
@kcreddy do you need to update the kibana constraint to ^8.18 || ^9.0 ?

kcreddy commented:
Moved to draft to avoid accidental merge.

qcorporation commented:
> > @kcreddy do you need to update the kibana constraint to ^8.18 || ^9.0 ?
> 
> @qcorporation, I have based my changes from [Security Team guide](https://github.com/elastic/security-team/blob/main/docs/cloud-security-team/agentless/onboard-integration.mdx). It doesn't suggest the version to target. Is there any suggestion you could provide what version to target? Among the very few integrations that have enabled `agentless`, I see [`cloud_security_posture`](https://github.com/elastic/integrations/blob/main/packages/cloud_security_posture/manifest.yml#L12-L14) has targeted `^8.18.0 || ^9.0.0`.
> 
> cc: @jamiehynds

@kcreddy Agentless is additional infrastructure that Elastic has to manage and I don't believe that it was offered until 8.17 specifically for CSPM for beta. So I don't know what would happen if we have a stack version less than 8.17/8.18 with the deployment_modes set to agentless what would happen.

kcreddy commented:
@qcorporation, thanks for clarification. I tested on `8.17.0` (earlier I used `8.17.1`) and it did seem to spin up the agentless infrastructure. 
I could go ahead with `^8.17.0 || ^9.0.0` because `8.17.0` is in GA while `8.18.0` is not. 
Are there any major improvements in agentless capabilities in `8.18` that make us choose `8.18.0` over `8.17.0`?

smriti0321 commented:
@kcreddy Yes there are changes in UI for 8.18, also state support. I would recommend using 8.18. cc @oren-zohar @olegsu @kpollich

qcorporation commented:
@kcreddy can you please see if it works with both inputs `azure-eventhub` ?

orestisfl commented on packages/microsoft_sentinel/manifest.yml:
```diff
@@ -1,7 +1,7 @@
 format_version: 3.2.1
```
@seanrathier this is the correct format version, right?
```suggestion
format_version: 3.2.3
```

https://github.com/elastic/package-spec/blob/964c4a69e024cc464c4808720ba0db9f001a82a7/spec/changelog.yml#L46-L53

seanrathier replied:
> @seanrathier this is the correct format version, right?

Yes, `3.2.3` is the most recent version that includes the `is_default` property. 

https://github.com/elastic/integrations/blob/main/packages/cloud_security_posture/manifest.yml#L110

seanrathier replied:
version 3.2.3 also included the agentless agent resource options.

https://github.com/elastic/package-spec/blob/main/spec/integration/manifest.spec.yml#L211

kcreddy replied:
Thank you! Updated to `3.2.3`

seanrathier reviewed (commented):
Overall pretty straightforward,  I  think you want to use the most recent version of the package spec at 3.2.3

jamiehynds commented:
> @kcreddy can you please see if it works with both inputs `azure-eventhub` ?

Good catch @qcorporation, I think agentless support is limited to CEL/httpjson at the moment, but Azure Event Hub is on the roadmap (@nimarezainia). 

We'll need to ensure it's clear in docs/UI that agentless support for Sentinel is limited to API and not supported for Event Hub.

qcorporation commented:
> > @kcreddy can you please see if it works with both inputs `azure-eventhub` ?
> 
> Good catch @qcorporation, I think agentless support is limited to CEL/httpjson at the moment, but Azure Event Hub is on the roadmap (@nimarezainia).
> 
> We'll need to ensure it's clear in docs/UI that agentless support for Sentinel is limited to API and not supported for Event Hub.

@kcreddy we should be testing on a 8.18 install. Within Kibana this input for agentless should not be available. If it is available then we to file a ticket with Kibana: https://github.com/elastic/kibana/issues/202091

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/22275)
* Commit: b388c013c7a9c33350d2bf92c5e45fa6dd784e92

### History
* :green_heart: [Build #21900](https://buildkite.com/elastic/integrations/builds/21900) succeeded 393c46a1a7cab5a896a50c6ec295ffdbd726e3cf
* :green_heart: [Build #21762](https://buildkite.com/elastic/integrations/builds/21762) succeeded f059f98581936ff4e75b49e6cfdbb4f60fd5ef01
* :green_heart: [Build #21746](https://buildkite.com/elastic/integrations/builds/21746) succeeded 384ba0aa4385e56775d1d0e167c96b046c3acd7d
* :green_heart: [Build #21568](https://buildkite.com/elastic/integrations/builds/21568) succeeded 26e5fdeaa77f369e7cc9b67cce0a89363f267aa2
* :broken_heart: [Build #21543](https://buildkite.com/elastic/integrations/builds/21543) failed 33a051c532f931067332dc21173dd53412f49c06

cc @kcreddy

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22275","number":22275,"commit":"b388c013c7a9c33350d2bf92c5e45fa6dd784e92"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21900","number":21900,"commit":"393c46a1a7cab5a896a50c6ec295ffdbd726e3cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21762","number":21762,"commit":"f059f98581936ff4e75b49e6cfdbb4f60fd5ef01"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21746","number":21746,"commit":"384ba0aa4385e56775d1d0e167c96b046c3acd7d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21568","number":21568,"commit":"26e5fdeaa77f369e7cc9b67cce0a89363f267aa2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21543","number":21543,"commit":"33a051c532f931067332dc21173dd53412f49c06"}],"number":22275}
buildkite-pr-comment-->

kcreddy commented:
> @kcreddy we should be testing on a 8.18 install. Within Kibana this input for agentless should not be available. If it is available then we need to file a ticket with Kibana as this should have been disabled by this ticket: https://github.com/elastic/kibana/issues/202091

@qcorporation, I created: https://github.com/elastic/kibana/issues/211092 for removing `Azure Eventhub` input from Agentless.

cc: @jamiehynds

kcreddy commented:
@qcorporation,
I was able to successfully test with API (CEL input) in both Cloud (8.18.0) and Serverless environments.

<details>
<summary>Successful ingestion in Cloud environment (8.18.0)</summary>
<img width="1691" alt="Screenshot 2025-02-13 at 5 45 09 PM" src="https://github.com/user-attachments/assets/c0dd73bf-c2f0-4bd9-9aeb-ea856e0a0780" />
</details>

<details>
<summary>Successful ingestion in Serverless environment</summary>
<img width="1721" alt="Screenshot 2025-02-13 at 5 14 15 PM" src="https://github.com/user-attachments/assets/74873737-8c14-4f26-ba55-8315529a3ece" />
</details>

nimarezainia commented:
@kcreddy thanks for this work. 
@qcorporation @kcreddy @smriti0321  what's the level of testing that will be performed on this integration? We have also been modeling the impact on COGs, at the moment portion of the cogs will depend on volume of traffic ingested. 

Seems that for this integration the default container sizes are used, which means that there may not be COGS related concern. However do we know that the default (1cpu, 1GiB) container is sufficient for MSFT sentinel?

I think as we start onboarding more integrations to agentless some level of stress testing would be good just to see if the sizing of the infrastructure is suitable. let me know your thoughts.

nimarezainia commented:
@kcreddy once the eventhub inout is blocked, what would the behavior be here for this integration? the MSFT sentinel events are to be ingested via Eventhub. if at the top level the integration is nominated to be agentless, would the user be allowed to enable collection of events?

kcreddy commented:
Moved the PR to draft once again as we are waiting on https://github.com/elastic/kibana/issues/211092.

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/22958)
* Commit: bb589a593ac4f38da0b3c396d3c3388d3be3c54c

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/22958#0195602c-a673-49f4-9566-aff8f3777bd2)

### History
* :green_heart: [Build #22275](https://buildkite.com/elastic/integrations/builds/22275) succeeded b388c013c7a9c33350d2bf92c5e45fa6dd784e92
* :green_heart: [Build #21900](https://buildkite.com/elastic/integrations/builds/21900) succeeded 393c46a1a7cab5a896a50c6ec295ffdbd726e3cf
* :green_heart: [Build #21762](https://buildkite.com/elastic/integrations/builds/21762) succeeded f059f98581936ff4e75b49e6cfdbb4f60fd5ef01
* :green_heart: [Build #21746](https://buildkite.com/elastic/integrations/builds/21746) succeeded 384ba0aa4385e56775d1d0e167c96b046c3acd7d
* :green_heart: [Build #21568](https://buildkite.com/elastic/integrations/builds/21568) succeeded 26e5fdeaa77f369e7cc9b67cce0a89363f267aa2
* :broken_heart: [Build #21543](https://buildkite.com/elastic/integrations/builds/21543) failed 33a051c532f931067332dc21173dd53412f49c06

cc @kcreddy

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22958","number":22958,"commit":"bb589a593ac4f38da0b3c396d3c3388d3be3c54c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22275","number":22275,"commit":"b388c013c7a9c33350d2bf92c5e45fa6dd784e92"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21900","number":21900,"commit":"393c46a1a7cab5a896a50c6ec295ffdbd726e3cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21762","number":21762,"commit":"f059f98581936ff4e75b49e6cfdbb4f60fd5ef01"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21746","number":21746,"commit":"384ba0aa4385e56775d1d0e167c96b046c3acd7d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21568","number":21568,"commit":"26e5fdeaa77f369e7cc9b67cce0a89363f267aa2"}],"number":22958}
buildkite-pr-comment-->

kcreddy commented:
@qcorporation, I updated the README as per suggestion from docs team.

`azure-eventhub` is still present in latest `8.18.0` snapshot. However it is now removed in `9.0.0` and also Serverless.

<details>
<summary>Azure Eventhub present in latest 8.18.0</summary>
<img width="813" alt="Screenshot 2025-03-04 at 1 25 40 PM" src="https://github.com/user-attachments/assets/8b630dcc-517b-4306-a97a-d9270b5aebd5" />
</details> 


<details>
<summary>Azure Eventhub removed in 9.0.0</summary>
<img width="806" alt="Screenshot 2025-03-04 at 1 25 24 PM" src="https://github.com/user-attachments/assets/9fd1b293-430d-42bc-9c85-5ebcd53c5a86" />
</details> 

<details>
<summary>Azure Eventhub removed in Serverless</summary>
<img width="813" alt="Screenshot 2025-03-04 at 1 37 40 PM" src="https://github.com/user-attachments/assets/2b86509a-2a74-4ac9-b7bf-b0b0074b4dcb" />
</details> 

It seems that the change removing Eventhub input: https://github.com/elastic/kibana/pull/211262 is merged into `9.0.0` and `9.1.0`. Should this be also backported to `8.18.0` or `8.19.0`?

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/22959)
* Commit: 12fefe4e404536071d7cf4522ca75b73713a2f59

### History
* :broken_heart: [Build #22958](https://buildkite.com/elastic/integrations/builds/22958) failed bb589a593ac4f38da0b3c396d3c3388d3be3c54c
* :green_heart: [Build #22275](https://buildkite.com/elastic/integrations/builds/22275) succeeded b388c013c7a9c33350d2bf92c5e45fa6dd784e92
* :green_heart: [Build #21900](https://buildkite.com/elastic/integrations/builds/21900) succeeded 393c46a1a7cab5a896a50c6ec295ffdbd726e3cf
* :green_heart: [Build #21762](https://buildkite.com/elastic/integrations/builds/21762) succeeded f059f98581936ff4e75b49e6cfdbb4f60fd5ef01
* :green_heart: [Build #21746](https://buildkite.com/elastic/integrations/builds/21746) succeeded 384ba0aa4385e56775d1d0e167c96b046c3acd7d
* :green_heart: [Build #21568](https://buildkite.com/elastic/integrations/builds/21568) succeeded 26e5fdeaa77f369e7cc9b67cce0a89363f267aa2

cc @kcreddy

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22959","number":22959,"commit":"12fefe4e404536071d7cf4522ca75b73713a2f59"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22958","number":22958,"commit":"bb589a593ac4f38da0b3c396d3c3388d3be3c54c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22275","number":22275,"commit":"b388c013c7a9c33350d2bf92c5e45fa6dd784e92"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21900","number":21900,"commit":"393c46a1a7cab5a896a50c6ec295ffdbd726e3cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21762","number":21762,"commit":"f059f98581936ff4e75b49e6cfdbb4f60fd5ef01"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21746","number":21746,"commit":"384ba0aa4385e56775d1d0e167c96b046c3acd7d"}],"number":22959}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/22974)
* Commit: 311b345c4f1e2b248c42b07fa99738706cb05d0d

### History
* :green_heart: [Build #22959](https://buildkite.com/elastic/integrations/builds/22959) succeeded 12fefe4e404536071d7cf4522ca75b73713a2f59
* :broken_heart: [Build #22958](https://buildkite.com/elastic/integrations/builds/22958) failed bb589a593ac4f38da0b3c396d3c3388d3be3c54c
* :green_heart: [Build #22275](https://buildkite.com/elastic/integrations/builds/22275) succeeded b388c013c7a9c33350d2bf92c5e45fa6dd784e92
* :green_heart: [Build #21900](https://buildkite.com/elastic/integrations/builds/21900) succeeded 393c46a1a7cab5a896a50c6ec295ffdbd726e3cf
* :green_heart: [Build #21762](https://buildkite.com/elastic/integrations/builds/21762) succeeded f059f98581936ff4e75b49e6cfdbb4f60fd5ef01
* :green_heart: [Build #21746](https://buildkite.com/elastic/integrations/builds/21746) succeeded 384ba0aa4385e56775d1d0e167c96b046c3acd7d

cc @kcreddy

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22974","number":22974,"commit":"311b345c4f1e2b248c42b07fa99738706cb05d0d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22959","number":22959,"commit":"12fefe4e404536071d7cf4522ca75b73713a2f59"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22958","number":22958,"commit":"bb589a593ac4f38da0b3c396d3c3388d3be3c54c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22275","number":22275,"commit":"b388c013c7a9c33350d2bf92c5e45fa6dd784e92"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21900","number":21900,"commit":"393c46a1a7cab5a896a50c6ec295ffdbd726e3cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21762","number":21762,"commit":"f059f98581936ff4e75b49e6cfdbb4f60fd5ef01"}],"number":22974}
buildkite-pr-comment-->

kcreddy commented:
Retested after https://github.com/elastic/kibana/pull/213104 in `8.18.0` and the Eventhub input is no longer visible in Agentless mode.

<img width="910" alt="Screenshot 2025-03-05 at 9 14 50 PM" src="https://github.com/user-attachments/assets/46e23bf1-393f-4a97-9506-66a525a93951" />
